homeassistant/components/systemmonitor/sensor.py

```diff
@@ -226,7 +226,9 @@ async def async_setup_platform(
             _LOGGER.warning("Cannot read CPU / processor temperature information")
             continue
 
-        sensor_registry[type_] = SensorData(argument, None, None, None, None)
+        if type_ not in sensor_registry:
+            sensor_registry[type_] = {}
+        sensor_registry[type_][argument] = SensorData(argument, None, None, None, None)
```

Comment:
How about using a tuple for the key instead of making the dict multi level

Ex: sensor_registry[(type_, argument)]

Reply:
Yes, that make sense- I’ll update to pull request tonight.